x-pack/test/case_api_integration/common/lib/utils.ts

```diff
@@ -154,34 +154,65 @@ export const createSubCase = async (args: {
   caseID?: string;
   caseInfo?: CasePostRequest;
   actionID?: string;
+  space?: string;
+  user?: User;
 }): Promise<CreateSubCaseResp> => {
   return createSubCaseComment({ ...args, forceNewSubCase: true });
 };
 
 /**
  * Add case as a connector
  */
-export const createCaseAction = async (supertest: st.SuperTest<supertestAsPromised.Test>) => {
-  const { body: createdAction } = await supertest
-    .post('/api/actions/connector')
+export const createCaseAction = async ({
+  supertest,
+  user,
+  space,
+}: {
+  supertest: st.SuperTest<supertestAsPromised.Test>;
+  user?: User;
+  space?: string;
+}) => {
+  const req = supertest
+    .post(`${getSpaceUrlPrefix(space)}/api/actions/connector`)
     .set('kbn-xsrf', 'foo')
     .send({
       name: 'A case connector',
       connector_type_id: '.case',
       config: {},
-    })
-    .expect(200);
+    });
+
+  const { body: createdAction } = await supertestAuth(req, user).expect(200);
   return createdAction.id;
 };
 
 /**
  * Remove a connector
  */
-export const deleteCaseAction = async (
-  supertest: st.SuperTest<supertestAsPromised.Test>,
-  id: string
-) => {
-  await supertest.delete(`/api/actions/connector/${id}`).set('kbn-xsrf', 'foo');
+export const deleteCaseAction = async ({
+  supertest,
+  id,
+  space,
+  user,
+}: {
+  supertest: st.SuperTest<supertestAsPromised.Test>;
+  id: string;
+  user?: User;
+  space?: string;
+}) => {
+  const req = supertest
+    .delete(`${getSpaceUrlPrefix(space)}/api/actions/connector/${id}`)
+    .set('kbn-xsrf', 'foo');
+  await supertestAuth(req, user);
+};
+
+/**
+ * Optionally authenticates if a user is supplied.
+ */
+const supertestAuth = (supertestInstance: supertestAsPromised.Test, user?: User) => {
```

Comment:
What do you think of using a function like this to avoid the `if (user)` check in each place we might need to call `auth`?

Reply:
What do you think about https://github.com/elastic/kibana/pull/97228/files/acb83b201e7f9fbb215a1043c2b79242f7bc50e5#r614582757

Reply:
Yeah I like that better 👍